File: src/main/java/com/dansoftware/libraryapp/util/PlatformName.java
package com.dansoftware.libraryapp.util;

import com.sun.javafx.PlatformUtil;

public class PlatformName {
    @Override
    public String toString() {
        if (PlatformUtil.isWindows()) return "Windows";
        else if (PlatformUtil.isLinux()) return "Linux";
        else if (PlatformUtil.isMac()) return "Mac";

        return System.getProperty("os.name");
    }
}

File: src/main/java/com/dansoftware/libraryapp/util/PlatformName.kt
package com.dansoftware.libraryapp.util

class PlatformName {
    override fun toString(): String = when {
        OsInfo.isWindows() -> "Windows"
        OsInfo.isLinux() -> "Linux"
        OsInfo.isMac() -> "Mac"
        else -> System.getProperty("os.name")
    }
}